### PR TITLE
Enable configuration to passive connection to exchange

### PR DESCRIPTION
**Pull request by lucasmarqs**
Ref #7 

I'm just validating whether the Listen should use a passive connection or a default one.

Plus I didn't figure out how to test this property in `amqp.Channel` struct since it has only _private_ properties.

### CODE REVIEW

File: rabbus.go [hunk 1]
```diff
@@ -303,3 +311,16 @@ func notifyClose(dsn string, r *rabbus) {
 		}
 	}
 }
+
+func declareExchange(ch *amqp.Channel, ex, kind string, config Config) error {
```
Comment (rafaeljesus) [inline rabbus.go]: It is fine for now, just as a info, hope soon I will declared this method to rabbus struct which will be renamed to smith else, so in the constructor wi'll have:
```go
type Rabbus interface {
 // methods omitted
}

func NewRabbus(c Config) (*RabbusAmqp, error) {
return &RabbusAmqp{}
}
```

This way we can declare private methods on the struct freely without the compiler complaining about method is not declared

File: rabbus.go [hunk 2]
```diff
@@ -303,3 +311,16 @@ func notifyClose(dsn string, r *rabbus) {
...
+	if config.PassiveExchange {
```
Comment (rafaeljesus) [inline rabbus.go]: What do you think of:

```go
func declareExchange(ch *amqp.Channel, ex, kind string, config Config) error {
  if config.PassiveExchange {
    return ch.ExchangeDeclarePassive(ex, kind, config.Durable, false, false, false, nil)
  }
  
  return ch.ExchangeDeclare(ex, kind, config.Durable, false, false, false, nil)
}
```

File: rabbus.go [hunk 3]
```diff
@@ -243,7 +250,8 @@ func (r *rabbus) register() {
 
 func (r *rabbus) produce(m Message) {
 	if _, ok := r.exDeclared[m.Exchange]; !ok {
-		if err := r.ch.ExchangeDeclare(m.Exchange, m.Kind, r.config.Durable, false, false, false, nil); err != nil {
+		err := declareExchange(r.ch, m.Exchange, m.Kind, r.config)
```
Comment (rafaeljesus) [inline rabbus.go]: WDYT of this:
```go
if err := declareExchange(r.ch, m.Exchange, m.Kind, r.config); err != nil {
  r.emitErr <- err
  return
}
```

**Comment by lucasmarqs** [comment]
@rafaeljesus I just did your requests. May you review it again? Tks! 😄

**Comment by rafaeljesus** [comment]
@lucasmarqs Thank you again!

**Pull request merged**